x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DataStreamTests.java

```diff
@@ -0,0 +1,225 @@
+        final AcknowledgedResponse downsampleResponse = client().admin()
+            .indices()
+            .execute(DownsampleAction.INSTANCE, downsampleRequest)
+            .actionGet();
+
+        client().admin()
+            .indices()
+            .updateSettings(
+                new UpdateSettingsRequest().indices(downsampleTargetIndex)
+                    .settings(Settings.builder().put(IndexMetadata.SETTING_INDEX_HIDDEN, false).build())
+            )
+            .actionGet();
+
+        final ModifyDataStreamsAction.Request modifyDataStreamRequest = new ModifyDataStreamsAction.Request(
+            List.of(
+                DataStreamAction.removeBackingIndex(dataStreamName, rolloverResponse.getOldIndex()),
+                DataStreamAction.addBackingIndex(dataStreamName, downsampleTargetIndex)
+            )
+        );
+        client().execute(ModifyDataStreamsAction.INSTANCE, modifyDataStreamRequest).actionGet();
+
+        // THEN
+        assertThat(downsampleResponse.isAcknowledged(), equalTo(true));
+        final GetDataStreamAction.Response getDataStreamActionResponse = client().admin()
+            .indices()
+            .execute(GetDataStreamAction.INSTANCE, new GetDataStreamAction.Request(new String[] { dataStreamName }))
+            .actionGet();
+        assertThat(getDataStreamActionResponse.getDataStreams().get(0).getDataStream().getIndices().size(), equalTo(2));
+        final List<String> backingIndices = getDataStreamActionResponse.getDataStreams()
+            .get(0)
+            .getDataStream()
+            .getIndices()
+            .stream()
+            .map(Index::getName)
+            .toList();
+        assertThat(backingIndices, Matchers.containsInAnyOrder(downsampleTargetIndex, rolloverResponse.getNewIndex()));
+
+        final SearchRequest searchRequest = new SearchRequest().indices(dataStreamName)
+            .source(
+                new SearchSourceBuilder().size(0)
+                    .query(new MatchAllQueryBuilder())
+                    .aggregation(
+                        new DateHistogramAggregationBuilder("dateHistogram").field("@timestamp").fixedInterval(DateHistogramInterval.MINUTE)
+                    )
+            );
+        final SearchResponse searchResponse = client().search(searchRequest).actionGet();
+        final InternalDateHistogram dateHistogram = searchResponse.getAggregations().get("dateHistogram");
+        // NOTE: due to unpredictable values for the @timestamp field we don't know how many buckets we have in the
+        // date histogram. We know, anyway, that we will have 10 documents in the first bucket, 10 documents in the last bucket
+        // and a variable number of intermediate buckets with exactly 0 documents. This is a result of the way downsampling deals
+        // with a fixed interval granularity that is larger than the date histogram fixed interval (1 minute (date histogram fixed_interval)
+        // < 1 hour (downsample fixed_interval)).
```

Comment:
Yes, here I am assuming that indexing a batch of 10 documents takes less than a minute...which I believe is a fair assumption...

Reply:
Can we run this search request before downsampling and then compare after downsampling that the results are the same as the results of the first search request?

Reply:
The original index (the one used as source index while doing downsampling) is removed from the data stream. That is because, at the moment, we still don't support multiple indices with the same time span, and we miss the logic to select the right index to use for each query.

Reply:
👍

Reply:
It looks like you are using a fixed timestamp for the generated documents. It looks like you are assuming that the 10 documents don't straddle the line. Maybe you should detect this and round up to the start of the next bucket or something. Or just be ok if they straddle in the assertion.

Reply:
I think I will test for something like this

```
sum(first and second bucket doc count) == 10 &&
sum(last two buckets doc count) == 0
sum(whatever is in between doc count) == 0
```